File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = [0] * len(photo)
    for i in range(len(photo)):
        for j in range(len(photo[i])):
            try:
                idx = name.index(photo[i][j])
                answer[i] += yearning[idx]
            except:
                continue
            
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.ArrayList;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        ArrayList<Integer> answer = new ArrayList<>();
        for (int i = 0; i < photo.length; i++) {
            int score = 0;
            for (int j = 0; j < photo[i].length; j++) {
                for (int k = 0; k < name.length; k++) {
                    if (photo[i][j].equals(name[k])) {
                        score += yearning[k];
                    }
                }
            }
            answer.add(score);
        }
        int[] result = answer.stream().mapToInt(Integer::intValue).toArray();
        return result;
    }
}